src/pages/Signup/Signup.js

```diff
@@ -1,8 +1,302 @@
+                        성별
+                        <select
+                          onChange={handleInput}
+                          name="gender"
+                          value={gender}
+                          type="select"
+                        >
+                          <option value={null} />
+                          <option value="male">남성</option>
+                          <option value="female">여성</option>
+                          <option value="deny">응답 거부</option>
+                        </select>
+                      </label>
+                    ) : (
+                      <React.Fragment key={id}>
+                        <label>
+                          {label}
+                          <input
+                            onChange={handleInput}
+                            value={inputs[name]}
+                            name={name}
+                            type={type}
+                            placeholder={placeholder}
+                          />
+                        </label>
+                        {validations[name] ? (
+                          <div className="okay">
+                            <CheckCircle className="icon" /> {messages[name][0]}
+                          </div>
+                        ) : (
+                          <div className="warning">
+                            <AlertCircle className="icon" /> {messages[name][1]}
+                          </div>
+                        )}
+                      </React.Fragment>
+                    )}
+                  </React.Fragment>
+                );
+              }
+            )}
+            <div className="checkbox-container">
+              <input
+                className="checkbox"
+                type="checkbox"
+                onChange={() =>
+                  setCheckboxes({
+                    ...checkboxes,
+                    terms: !terms,
+                  })
+                }
+                value={terms}
+              />
+              <span className="label">
+                (필수) 약관을 모두 읽고 동의합니다. 이용약관
+              </span>
+            </div>
+            <div className="checkbox-container">
+              <input
+                className="checkbox"
+                type="checkbox"
+                onChange={() =>
+                  setCheckboxes({
+                    ...checkboxes,
+                    privacy: !privacy,
+                  })
+                }
+                value={privacy}
+              />
+              <span className="label">
+                (필수) 개인정보 수집ㆍ이용에 동의합니다. 개인정보 수집ㆍ이용
+                동의
+              </span>
+            </div>
+            <button
+              className={`submit-btn ${!allValueValidated && 'disabled'}`}
+              onClick={handleSubmit}
+            >
+              회원 가입하기
+            </button>
+          </form>
+        </div>
+      </div>
+    </div>
+  );
+};
 
 export default Signup;
+
```

Comment:
```suggestion
const SIGNUP_IMAGE_LIST = [
  {
    id: 1,
    img: [
      { id: 1, type: 'reg' },
      { id: 2, type: 'reg' },
      { id: 3, type: 'short' },
      { id: 4, type: 'reg' },
    ],
  },
  {
    id: 2,
    img: [
      { id: 1, type: 'short' },
      { id: 2, type: 'short' },
      { id: 3, type: 'reg' },
      { id: 4, type: 'short' },
      { id: 5, type: 'reg' },
    ],
  },
];
```

signup 페이지에 있는 이미지 관련해서 이 부분도 상수데이터로 구현할 수 있을 것 같네요!

Reply:
추가로 이렇게 상수데이터를 구현하게 되면 상수데이터 부분의 코드가 너무 길어지기 때문에
상수데이터만 관리하는 파일을 따로 생성하셔서 import 하는 방식으로 사용하시면 될 것 같습니다!